airbyte-integrations/connectors/source-brightspace/source_brightspace/api.py

```diff
@@ -57,22 +57,70 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def token_manager(func):
+    """Decorator to manage tokens for API access."""
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        # Check if access token exists
+        if not self.access_token or not self.valid_token():
+            # Generate a new token if it doesn't exist or is invalid
+            self.get_token()
+            self.logger.info("Generated a new access token.")
+
+        # Call the original function with updated token
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
 class BrightspaceClient:
     logger = logging.getLogger("airbyte")
     version = "1.43"
 
+    def valid_token(self) -> bool:
+        try:
+            url = f"{self.url_base}/dataExport/list"
+            headers = {"Authorization": "Bearer {}".format(self.access_token)}
+            _ = self._make_request(http_method="GET", url=url, headers=headers)
+        except HTTPError as error:
+            # if error.response.status_code == codes.UNAUTHORIZED:
+            self.logger.info(f"Unauthorized token received. Access token is invalid.")
+            return False
+            # raise error
+        return True
+
+    def get_token(self):
+        payload = {
+            'source_id': self.source_id,
```

Comment:
I think this should be part of your url as param

Reply:
removed